dspace-api/src/main/java/org/dspace/storage/rdbms/migration/V5_7_2017_05_05__DS_3431_Add_Policies_for_BasicWorkflow.java

```diff
@@ -7,45 +7,42 @@
  */
 package org.dspace.storage.rdbms.migration;
 
-import java.sql.Connection;
-
-import org.dspace.core.Constants;
 import org.dspace.storage.rdbms.DatabaseUtils;
-import org.flywaydb.core.api.migration.MigrationChecksumProvider;
-import org.flywaydb.core.api.migration.jdbc.JdbcMigration;
-import org.flywaydb.core.internal.util.scanner.classpath.ClassPathResource;
+import org.flywaydb.core.api.migration.BaseJavaMigration;
+import org.flywaydb.core.api.migration.Context;
+import org.springframework.core.io.ClassPathResource;
 
 public class V5_7_2017_05_05__DS_3431_Add_Policies_for_BasicWorkflow
-    implements JdbcMigration, MigrationChecksumProvider {
+    extends BaseJavaMigration {
 
     // Size of migration script run
     Integer migration_file_size = -1;
 
 
     @Override
-    public void migrate(Connection connection) throws Exception {
+    public void migrate(Context context) throws Exception {
         // Based on type of DB, get path to SQL migration script
-        String dbtype = DatabaseUtils.getDbType(connection);
+        String dbtype = DatabaseUtils.getDbType(context.getConnection());
 
         String dataMigrateSQL;
         String sqlMigrationPath = "org/dspace/storage/rdbms/sqlmigration/workflow/" + dbtype + "/";
         // Now, check if the XMLWorkflow table (cwf_workflowitem) already exists in this database
         // If XMLWorkflow Table does NOT exist in this database, then lets do the migration!
         // If XMLWorkflow Table ALREADY exists, then this migration is a noop, we assume you manually ran the sql
         // scripts
-        if (DatabaseUtils.tableExists(connection, "cwf_workflowitem")) {
+        if (DatabaseUtils.tableExists(context.getConnection(), "cwf_workflowitem")) {
             return;
         } else {
             //Migrate the basic workflow
             // Get the contents of our data migration script, based on path & DB type
-            dataMigrateSQL = new ClassPathResource(sqlMigrationPath + "basicWorkflow" + "/V5.7_2017.05.05__DS-3431.sql",
-                                                   getClass().getClassLoader())
-                .loadAsString(Constants.DEFAULT_ENCODING);
+            dataMigrateSQL = MigrationUtils.resourceToString(
+                new ClassPathResource(sqlMigrationPath + "basicWorkflow" + "/V5.7_2017.05.05__DS-3431.sql",
```

Comment:
I'm a bit worried about depending on a Flyway internal interface, and puzzled as to why we use it (`ClassPathResource`).

Reply:
@mwoodiupui : I'm not sure if I understand your comment?  Are you saying that `MigrationUtils.resourceToString()` should be changed to just take in a resource path?  I guess that's reasonable... I had simply done it this way so that I could use `Resource.getInputStream()` to easily read the resource....though, I guess I could have refactored all this to use the ClassLoader instead...e.g. using something like `ClassLoader.getResourceAsStream(String path)`.

It doesn't much change what the code does here, as it's essentially the same.  But, if I understand your comment, you are right that it might simplify things to refactor out the usage of `ClassPathResource` altogether.  Is that what you are referring to?

Reply:
Yes.  ClassPathResource can do a number of things, but the only thing it is doing for us is holding a reference to a String.  We may as well pass the String reference to `resourceToString` directly, instead of wrapping it in another object.  I wouldn't refuse to approve this patch as-is just because of this, though.